### PR TITLE
added syn:tag:part type and syn:tag norm handler for list/tuple

### PR DESCRIPTION
**Pull request by invisig0th**

### CODE REVIEW

File: synapse/tests/test_lib_storm.py [hunk 1]
```diff
@@ -506,6 +506,21 @@ async def skipone(n):
                     with self.raises(s_exc.StormRuntimeError):
                         await runt.getOneNode('ou:org:name', 'dupcorp')
 
+            opts = {'vars': {'tag': ('foo', 'part')}}
+            self.len(1, await core.nodes('[syn:tag=$tag]', opts=opts))
+            self.len(1, await core.nodes('syn:tag=$tag', opts=opts))
+
+            nodes = await core.nodes('[ inet:ipv4=1.2.3.4 +#$tag ]', opts=opts)
```
Comment (vEpiphyte) [inline synapse/tests/test_lib_storm.py:513]: This breaks existing behavior

```
cli> storm --optsfile opts.json $lib.print($strlist) [inet:ipv4=1.2.3.4 +#$strlist]
Executing query at 2021/05/14 00:45:12.072
('foo', 'bar', 'baz')
inet:ipv4=1.2.3.4
        .created = 2021/05/14 00:44:21.217
        :type = unicast
        #bar
        #baz
        #foo
complete. 1 nodes in 32 ms (31/sec).
```

Comment (vEpiphyte) [reply synapse/tests/test_lib_storm.py:513]: see #1300

File: synapse/tests/test_lib_types.py [hunk 2]
```diff
@@ -837,12 +840,14 @@ def test_syntag(self):
...
         self.raises(s_exc.BadTypeValu, tagtype.norm, 'foo\u202e.bar')
         self.raises(s_exc.BadTypeValu, tagtype.norm, 'foo.')
+        self.raises(s_exc.BadTypeValu, tagtype.norm, 'foo..bar')
         self.raises(s_exc.BadTypeValu, tagtype.norm, '.')
         self.raises(s_exc.BadTypeValu, tagtype.norm, '')
         # Tags including non-english unicode letters are okay
         self.eq('icon.ॐ', tagtype.norm('ICON.ॐ')[0])
         # homoglyphs are also possible
         self.eq('is.ｂob.evil', tagtype.norm('is.\uff42ob.evil')[0])
+        self.eq('icon.ॐ', tagtype.norm('ICON.ॐ')[0])
```
Comment (vEpiphyte) [inline synapse/tests/test_lib_types.py:850]: is this a duplicated of line 847?

Comment (invisig0th) [reply synapse/tests/test_lib_types.py]: yep! removed :D

File: synapse/lib/types.py [hunk 3]
```diff
@@ -1602,27 +1602,47 @@ class Tag(Str):
...
 
+    def _normPyStr(self, text):
+        toks = text.strip('#').split('.')
+        return self._normPyList(toks)
+
+class TagPart(Str):
+
+    def postTypeInit(self):
+        Str.postTypeInit(self)
+        self.setNormFunc(str, self._normPyStr)
```
Comment (vEpiphyte) [inline synapse/lib/types.py:1635]: This needs to accept norming a tuple

**Review by vEpiphyte — APPROVED** [review]

**Pull request merged**